exporter/tanzuobservabilityexporter/metrics_test.go

```diff
@@ -197,6 +196,270 @@ func TestGaugeConsumerMissingValue(t *testing.T) {
 	assert.Len(t, allLogs, expectedMissingValueCount)
 }
 
+// Tests that the histogramConsumer correctly delegates to its
+// histogramDataPointConsumers. This tests delta histograms
+func TestHistogramConsumerDeltaAggregation(t *testing.T) {
+	// TODO: Change this test when delta histograms are supported.
+	countAttributeForEachDataPoint := []uint64{2, 5, 10}
+	deltaMetric := newHistogramMetricWithDataPoints(
+		"delta.metric",
+		pdata.MetricAggregationTemporalityDelta,
+		countAttributeForEachDataPoint)
+	sender := &mockGaugeSender{}
+	cumulativeConsumer := &mockHistogramDataPointConsumer{}
+	consumer := newHistogramConsumer(
+		cumulativeConsumer, nil, sender, nil)
+	var errs []error
+	consumer.Consume(deltaMetric, &errs)
+	assert.Len(t, errs, 1)
+	consumer.PushInternalMetrics(&errs)
+	assert.Len(t, errs, 1)
+}
+
+// Tests that the histogramConsumer correctly delegates to its
+// histogramDataPointConsumers. This tests cumulative histograms
+func TestHistogramConsumerCumulativeAggregation(t *testing.T) {
+	countAttributeForEachDataPoint := []uint64{2, 5, 10}
+	cumulativeMetric := newHistogramMetricWithDataPoints(
+		"cum.metric",
+		pdata.MetricAggregationTemporalityCumulative,
+		countAttributeForEachDataPoint)
+	sender := &mockGaugeSender{}
+	cumulativeConsumer := &mockHistogramDataPointConsumer{}
+	consumer := newHistogramConsumer(
+		cumulativeConsumer, nil, sender, nil)
+	var errs []error
+
+	consumer.Consume(cumulativeMetric, &errs)
+	consumer.PushInternalMetrics(&errs)
+
+	assert.Empty(t, errs)
+
+	// We had three datapoints. Our mock just captures the metric name of
+	// each data point consumed.
+	assert.Equal(
+		t, []string{"cum.metric", "cum.metric", "cum.metric"}, cumulativeConsumer.names)
+	assert.Equal(t, countAttributeForEachDataPoint, cumulativeConsumer.counts)
+}
+
+// This tests that the histogram consumer correctly counts and logs
+// histogram metrics with missing aggregation attribute.
+func TestHistogramConsumerNoAggregation(t *testing.T) {
+
+	// Create a histogram metric with missing aggregation attribute
+	metric := newHistogramMetricWithDataPoints(
+		"bad.metric", pdata.MetricAggregationTemporalityUnspecified, nil)
```

Comment:
Might be nice for future-proofing to name this metric based on _why_ it's bad. `missing.aggregation.metric` or something?

Reply:
Done.